crimsobot/cogs/cringo.py

```diff
@@ -1,225 +1,423 @@
+            descr='\n'.join([
+                'Match the emojis called to the emojis on your card.',
+                'If you see a match, type the column and row of the match!',
+                'Type `.<letter><number>` or `. <letter><number>`.',
+                'You can put in multiple matches separated by a space!',
+                'For example: `.a1 b2 c4` or `. b4 c3`. Only use one period!',
+                'Missed a match on a previous turn? No problem! Put it in anyway.',
+                "You'll still get your points (but with a lower multiplier).",
+                'Need to leave the game? Type `.leave` during a round.',
+            ]),
+            thumb_name='jester'
+        )
+
+        return embed
+
+    def generate_join_no_dms_embed(self, who: discord.Member) -> discord.Embed:
+        embed = c.crimbed(
+            title=f'Uh oh, **{who} CANNOT** join the game!',
+            descr='You have to be able to receive DMs from crimsoBOT to play!',
+            color_name='orange',
+            thumb_name='weary'
+        )
+
+        return embed
+
+    def generate_join_low_balance_embed(self, who: discord.Member) -> discord.Embed:
+        embed = c.crimbed(
+            title=f'Uh oh, **{who} CANNOT** join the game!',
+            descr='\n'.join([
+                '· You must have a balance of \u20A2{:.2f} to play this game!'
+                .format(CRINGO_RULES['minimum_balance'][self.card_size]),
+                "· Don't fret; regular CRINGO! can be played by anyone!",
+            ]),
+            color_name='orange',
+            thumb_name='moneymouth'
+        )
 
-        # initialize player objects
-        list_of_players = []
-        for player in join_results.joined:
-            player_object = cringo.CringoPlayer()
-            player_object.user = player
-            player_object.card = await cringo.cringo_card(await cringo.cringo_emoji(card_size, card_size))
-            list_of_players.append(player_object)
+        return embed
+
+    def generate_join_already_playing_embed(self, who: discord.Member) -> discord.Embed:
+        embed = c.crimbed(
+            title=f'Uh oh, **{who} CANNOT** join the game!',
+            descr="You're already playing another game of CRINGO! aren't you?",
+            color_name='orange',
+            thumb_name='think'
+        )
+
+        return embed
+
+    def generate_join_success_embed(self, who: discord.Member) -> discord.Embed:
+        embed = c.crimbed(
+            title='',
+            descr=f'**{who}** has joined the game!',
+            color_name='green',
+            thumb_name=''
+        )
+
+        return embed
+
+    def generate_mismatch_embed(self) -> discord.Embed:
+        embed = c.crimbed(
+            title=None,
+            descr='Mismatch(es) detected. You lose points for that!',
+            color_name='orange',
+        )
+
+        return embed
+
+    def remove_from_game(self, user: discord.User) -> discord.Embed:
```

Comment:
this isn't actually removing the user from the game? just sending the embed. looks like an omission

Reply:
yeah it is an omission, my bad